examples/src/authentication/client.rs

```diff
@@ -9,7 +9,7 @@ use tonic::{metadata::MetadataValue, transport::Channel, Request};
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let channel = Channel::from_static("http://[::1]:50051").connect().await?;
 
-    let token = MetadataValue::from_str("Bearer some-auth-token")?;
+    let token: MetadataValue<_> = "Bearer some-auth-token".parse()?;
```

Comment:
hmm I actually kinda prefer the look of `from_str` vs using `parse`. I wonder if we should _not_ deprecate it and keep it. But just have the TryFrom's use it or the other way around?

Reply:
Hmm. So, MetadataValue impls `FromStr`, so it will already have a `from_str` method from stdlib, which is equivalent to `parse`. Unfortunately it's not in the stdlib prelude, so if I delete the ad-hoc `from_str` method then old code will fail to compile (although the compiler will suggest the easy fix of "use std::str::FromStr").

It does _kinda_ annoy me to have an adhoc method named `from_str` when the stdlib one exists too, just because it seems to make the API a bit larger than it needs to be. But this is a _very_ minor gripe :) So overall I'm happy with the solution of not deprecating `from_str` if that's what you'd like.

(I'm pretty used to `parse()` because it's used very often in the [stdlib IP types](https://doc.rust-lang.org/stable/std/net/struct.Ipv4Addr.html#examples)).

So yeah, happy to do whichever, just let me know what you think now that you've ready this silly long reply :)